events/keywords.py

```diff
@@ -1,112 +1,78 @@
-import re
-from events.models import Keyword, KeywordLabel, DataSource
-from difflib import get_close_matches
+from django.conf import settings
+from django.contrib.postgres.search import SearchQuery, TrigramSimilarity
+from rest_framework.exceptions import ParseError
 
-ENDING_PARENTHESIS_PATTERN = r' \([^)]+\)$'
+from events.models import KeywordLabel
 
 
 class KeywordMatcher(object):
     def __init__(self):
-        label_to_keyword_ids = {}
-        self.name_to_keyword_ids = {}
-        for label_id, keyword_id in Keyword.alt_labels.through.objects.all().values_list(
-                'keywordlabel_id', 'keyword_id'):
-            label_to_keyword_ids.setdefault(label_id, set()).add(keyword_id)
-        for label_id, name in KeywordLabel.objects.filter(language_id='fi').values_list(
-                'id', 'name'):
-            self.name_to_keyword_ids[name.lower()] = label_to_keyword_ids.get(label_id, set())
-        try:
-            yso_source = DataSource.objects.get(pk='yso')
-            self.skip = False
-        except DataSource.DoesNotExist:
-            print('No YSO keyword data source')
-            self.skip = True
-            return
-        for kid, preflabel in Keyword.objects.filter(data_source=yso_source).values_list(
-                'id', 'name_fi'):
-            if preflabel is not None:
-                text = preflabel.lower()
-                self.name_to_keyword_ids.setdefault(text, set()).add(kid)
-                without_parenthesis = re.sub(ENDING_PARENTHESIS_PATTERN, '', text)
-                if without_parenthesis != text:
-                    self.name_to_keyword_ids.setdefault(without_parenthesis, set()).add(kid)
-        self.labels = self.name_to_keyword_ids.keys()
-        print('Initialized', len(self.labels), 'keyword keys')
+        pass
 
-    def match(self, text):
-        if self.skip:
-            return None
-        wordsplit = re.compile(r'\s+')
-        # labels = KeywordLabel.objects
-        # match = labels.filter(name__iexact=text)
+    def full_text_matching(self, text, language=None):
```

Comment:
This method has good comments, but it definitely needs some unit tests. It would not only to make it even clearer after comments, but would also be a proof that it really works.